src/app/core/mods/showResources/resources.ts

```diff
@@ -0,0 +1,105 @@
+export class Resources {
+    private name: string;
+    private typeId: number;
+    private resources: {elemId: number, state: State}[] = [];
+
+    constructor(resource: Resource) {
+        this.name = resource.name;
+        this.typeId = resource.elementTypeId;
+        this.addOrUpdateResource(resource);
+    }
+
+    public addOrUpdateResource(resource: Resource) {
+        if (this.resources.length > 0 && this.resources.find(r => r.elemId == resource.elementId) != undefined) {
```

Comment:
`if (this.resources.length && this.resources.find(r => r.elemId == resource.elementId)) {`

Reply:
Event better 

```
const ref = this.resources.find(r => r.elemId == resource.elementId);
if (ref) ref.state = resource.elementState;
else this.resources.push({elemId: resource.elementId, state: resource.elementState})
```